source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -734,11 +734,15 @@ The event API here is assumed to be like the standard `Python Event
                 # Initiate streaming hello or legacy hello
                 if connectTimeoutMS != 0:
                     set connection timeout to connectTimeoutMS+heartbeatFrequencyMS
-                response = call {helloCommand: 1, topologyVersion: previousDescription.topologyVersion, maxAwaitTimeMS: heartbeatFrequencyMS}
+                response = call {helloCommand: 1, helloOk: True, topologyVersion: previousDescription.topologyVersion, maxAwaitTimeMS: heartbeatFrequencyMS}
             else:
                 # The server does not support topologyVersion.
-                response = call {helloCommand: 1}
+                response = call {helloCommand: 1, helloOk: True}
 
+            # if server supports hello, then response.helloOk will be true
+            # and hello will be used for subsequent monitoring commands.
+            # if server does not support hello, then response.helloOk will be undefined
+            # and legacy hello will be used for subsequent monitoring commands.
```

Comment:
To avoid adding `helloOk` to the ServerDescription:
```suggestion
            # and legacy hello will be used for subsequent monitoring commands.
            helloOk = response.helloOk
```

Reply:
I had added `helloOk` to `ServerDescription`:
https://github.com/JamesKovacs/specifications/blob/drivers1293/sdam/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#serverdescription

But thinking about it more, I believe we can implement this using a `helloOk` field in `Monitor` and `RttMonitor` thus not requiring changes to the public API for `ServerDescription`. The C# driver exposes `ServerDescription` in our eventing API and I suspect many/all other drivers do so as well. So better not to modify the public API for `ServerDescription` if we don't need to.